draft-thomson-http-oblivious.md

```diff
@@ -264,15 +306,131 @@ Message encrypted message.
+able to use that information to link requests. Cookies {{?COOKIES=RFC6265}} are
+the most obvious feature that MUST NOT be used by clients. However, clients
+need to include all information learned from requests, which could include the
+identity of resources.
+
+Clients also need to ensure that they correctly generate a new HPKE context for
+every request, using a good source of entropy ({{?RANDOM=RFC4086}}). Key reuse
+not only risks linkability, but it could expose request and response contents
+to the proxy. Note that an HPKE implementation is stateful and so prevents AEAD
+nonce reuse.
+
+Clients constructing the request that is to be encapsulated need to avoid
+including identifying information. Similarly, the request that is sent to the
+oblivious request resource, though this request can contain only minimal
+information as it only needs to include a method and the oblivious request
+resource URL.
+
+
+## Proxy Responsibilities
+
+The proxy that serves the oblivious proxy resource has a very simple function
+to perform. It forwards messages received at this resource to the oblivious
+request resource.
+
+The proxy MUST NOT add information about the client identity when forwarding
+requests. This includes the Via field, the Forwarded field
+{{?FORWARDED=RFC7239}}, and any similar information.
+
+
+### Denial of Service {#dos}
+
+As there are privacy benefits from having a large rate of requests forwarded by
+the same proxy (see {{ta}}), servers that operate the oblivious request
+resource might need an arrangement with proxies. This arrangement might be
+necessary to prevent having the large volume of requests being classified as an
+attack by the server.
+
+If a server does accept a large volume of requests from a proxy, it needs to
+trust that the proxy does not allow abusive levels of request volumes from
+clients. That is, if a server allows requests from the proxy to be exempt from
+rate limits, the server might want to ensure that the proxy applies similar
+rate limiting when receiving requests from clients.
+
+Servers that enter into an agreement with a proxy that enables a higher request
+rate might choose to authenticate the proxy to enable the higher rate.
+
+
+### Linkability Through Traffic Analysis {#ta}
+
+As the time at which encapsulated request or response messages are sent can
+reveal information to a network observer. Though messages exchanged between the
+oblivious proxy resource and the oblivious request resource might be sent in a
+single connection, traffic analysis could be used to match messages that are
+forwarded by the proxy.
+
+A proxy could, as part of its function, add delays in order to increase the
+anonymity set into which each message is attributed. This could latency to the
+overall time clients take to receive a response, which might not what some
+clients want.
+
+A proxy can use padding to reduce the effectiveness of traffic analysis.
```

Comment:
Noting transport-layer since it can't modify the binary messages.

```suggestion
A proxy can use transport-layer padding to reduce the effectiveness of traffic analysis.
```